### PR TITLE
New test for Lie Group

### PR DESCRIPTION
**Pull request by SaitejaUtpala**
## Description 

adds `test_to_tangent_at_identity_belongs_to_lie_algebra` to lie groups

### CODE REVIEW

File: tests/geometry_test_cases.py [hunk 1]
```diff
@@ -286,6 +286,25 @@ def test_exp_after_log(self, group_args, point, base_point, rtol, atol):
...
+    def test_to_tangent_at_identity_belongs_to_lie_algebra(
+        self, group_args, vec, belongs_atol
+    ):
+        """Check that to tangent at identity is tangent in lie algebra.
+
+        Parameters
+        ----------
+        group_args : tuple
+            Arguments to pass to constructor of the group.
+        tangent_vec : array-like
```
Comment (ninamiolane) [inline tests/geometry_test_cases.py:298]: adapt docstring

**Review by ninamiolane — APPROVED** [review]
Good to merge once the docstring is corrected

File: tests/data_generation.py [hunk 2]
```diff
@@ -471,6 +471,29 @@ def _exp_after_log_test_data(
...
         return self.generate_tests(smoke_data, random_data)
 
+    def _to_tangent_at_identity_belongs_to_lie_algebra_test_data(
+        self, group_args_list, shape_list, n_vecs_list, belongs_atol=gs.atol
+    ):
+        """Generate data to check that to tangent at identity belongs to lie algebra.
+
+        Parameters
+        ----------
+        group_args_list : _type_
```
Comment (ninamiolane) [inline tests/data_generation.py:481]: the argument `shape_list` is missing from the docstring. Please make sure that the docstrings are correct before sending the PR to review 🙏

**Review by ninamiolane — CHANGES_REQUESTED** [review]
Docstrings

File: tests/data_generation.py (hunk 2 above)
Comment (SaitejaUtpala) [reply tests/data_generation.py]: Omg! really sorry. this will not repeat from next time for sure!

**Review by ninamiolane — APPROVED** [review]
thanks

**Pull request merged**